src/Integration.Vsix/Delegates/SonarAnalyzerStandaloneWorkflow.cs

```diff
@@ -0,0 +1,95 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Helpers;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class SonarAnalyzerStandaloneWorkflow : SonarAnalyzerWorkflowBase
+    {
+        private readonly Func<IAnalysisRunContext, bool> previousShouldExecuteRuleFunc;
+        private readonly Action<IReportingContext> previousReportDiagnosticAction;
+
+        public SonarAnalyzerStandaloneWorkflow(Workspace workspace)
+            : base(workspace)
+        {
+            // Saving previous state so that SonarLint doesn't have to know what's the default state in SonarAnalyzer
```

Comment:
Could this save-and-reset code be in the base class?

Reply:
Standalone and Connected workflows use the same delegates but not LegacyConnected.